File: dm-frontend/faultyDevice/FaultyDeviceModel.ts
export class FaultyDeviceModel
{
    public complaintId : number
    public userId : number  
    public userName : string
    public deviceId : number 
    public deviceType : string
    public deviceName : string 
    public serialNumber : string 
    public status : string
    public complaitDate  : string 
    public complaint : string


    constructor(data : any )
    {
        this.complaintId = data.complaintId;
        this.userId = data.userId;
        this.deviceId = data.deviceId;
        this.serialNumber= data.serialNumber;
        this.userName = data.salutation  + " " + data.userName.first_name + " " +( data.userName.middle_name== "" ? "": data.userName.middle_name + " "  ) +data.userName.last_name;
        this.deviceType = data.deviceType; 
        this.deviceName =  data.deviceBrand +" " + data.deviceModel ; 
        this.status = data.status;
        this.complaint = data.issue;
        this.complaitDate = data.complaintDate;
    }
}

File: dm-backend/Models/FaultyDeviceModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace dm_backend.Models
{
    public class FaultyDeviceModel
    {
        public int complaintId { get; set; }
        public int deviceId { get; set; }
        public string deviceType {get; set;}
        public string deviceBrand { get; set; }
        public string deviceModel { get; set; }
        public string serialNumber { get; set; }
        public int userId { get; set; }
        public string salutation { get; set; }
        public name userName { get; set; }
        public string complaintDate { get; set; }
        public string Issue { get; set; }
        public string status { get; set; }

    }
}
